packages/eslint-plugin/src/rules/no-misused-promises.ts

```diff
@@ -568,11 +571,27 @@ export default createRule<Options, MessageId>({
           continue;
         }
 
+        if (
+          node.type !== AST_NODE_TYPES.MethodDefinition &&
+          node.type !== AST_NODE_TYPES.TSAbstractMethodDefinition &&
+          node.type !== AST_NODE_TYPES.TSMethodSignature &&
+          node.type !== AST_NODE_TYPES.TSPropertySignature &&
+          node.type !== AST_NODE_TYPES.PropertyDefinition
```

Comment:
This is enough to pass the test suit but there are additional types available to satisfy `getStaticMemberAccessValue`.

Reply:
This is necessary mainly for type-checking, and I don't know if this can realistically have a runtime scenario. Please let me know if this should be handled differently (currently, there is no coverage for this).

Reply:
I don't believe this can realistically be a runtime scenario. I think the real issue here is that although `nodeMember` is type `ts.TypeElement | ts.ClassElement`, `node` is type `TSESTree.Node`. 

The two strategies I'd think could be good would be:

* Simpler: use an `as` to correct TypeScript's understanding of `node`'s type to be `NodeWithKey` (which will need to be exported from `misc.ts`)
* More aligned with how most of our typed lint rules go: instead of iterating over the `tsNode.members`, iterate over the members of `node`

The latter is probably a lot more work and changes the existing code. So I think the `as` strategy would be reasonable.

Reply:
Yes, I've started with a type-assertion on my initial implementation but I wasn't sure if there's a realistic case where this would explode.

I updated the code to use a type assertion, as the second option seems to be a lot more work.